docs: add filter demos

kmcfaul opened the pull request:
Tied to #6710 but will not fully close the issue (more demos are requested). Since the groundwork of updating older demos to use the newer composable components and putting the demo into TSX/functional form was a bulk of the work and is now done the rest should be quicker to finish, I'm just not positive I'll get them all out in this PR.

Working off the design here: https://marvelapp.com/prototype/7i0dj2d/screen/75593646

- [x] Search input
- [x] Single select
- [x] Checkbox select
- [ ] Attribute filter
- [ ] Filter group
- [ ] Faceted filter

Demos are under the Demos section in the "Filters" page. Let me know if they should be moved elsewhere or called something else.

kmcfaul commented:
@mcarrano @mmenestr I have a question regarding the marvel proposal about the purpose of the checkbox and pagination in the toolbar section. The table already has a bulk selection element in the header, and does not have enough data to require pagination, so are these elements required? Should they be present regardless without any function?

patternfly-build commented:
Preview: https://patternfly-react-pr-7949.surge.sh

mmenestr commented:
@kmcfaul I would say the pagination should be presented still but disabled since there is no other page to view. And for the bulk select, I would favor having a bulk select in the toolbar over the table header!

kmcfaul commented:
@mmenestr Ok I can work on those updates. What should be options be for the bulk select dropdown then? `Select all`, `Deselect all`?

kmcfaul commented:
Went with those two options for now.

Also fixed a small bug where `onClick` wasn't being piped through to the split button variant of `MenuToggle`.

thatblindgeye commented on packages/react-core/src/demos/Filters/examples/FilterCheckboxSelect.tsx:
```diff
@@ -0,0 +1,394 @@
+
+  // Set up pagination
+  const toolbarPagination = (
+    <Pagination
+      perPageComponent="button"
+      itemCount={repositories.length}
+      perPage={50}
+      page={1}
+      widgetId="table-mock-pagination"
```
To resolve an aXe issue, `paginationTitle` should be passed in with a unique title so that the `aria-label` is unique across the three demos thus far

thatblindgeye commented on packages/react-core/src/demos/Filters/examples/FilterCheckboxSelect.tsx:
```diff
@@ -0,0 +1,394 @@
+  );
+
+  const toolbar = (
+    <Toolbar id="toolbar-items" clearAllFilters={() => setSelections([])}>
```
Similar to above, the id just needs to be unique across the three demos

thatblindgeye reviewed (changes requested):
In addition to comments below, aXe is flagging the skip in heading levels (h1 to h3). Adding an h2 "Demos" heading similar to other demo pages would resolve it.

kmcfaul commented:
Updated for the aXe issues.

mmenestr commented:
@kmcfaul would go with these options to match other demos in the table pages: 
<img width="229" alt="image" src="https://user-images.githubusercontent.com/51165119/189423840-4a597cd2-0b4b-4df9-a785-d7f6ec6c0f35.png">

And for pagination can we do compact pagination actually? Sorry about that

mcarrano reviewed (changes requested):
Besides @mmenestr 's comments, here are a couple more things I'm noticing:

* When the filter results in no items being returned, we should show an empty state for the table. The one to use would be here: https://www.patternfly.org/v4/components/empty-state#no-match-found
* In testing the bulk-select action, the checkbox should be showing mixed state (-) when only some items are selected.

kmcfaul commented:
@mmenestr Updated the bulk select options and made the pagination compact. I used 10 per page and 10 total results to match what the table has and what the pagination is showing. Let me know if you would rather the bulk select and pagination show a 20 per page/100 total setup instead of using the 10 current entries.

@mcarrano 
Added the empty state handling in for the search input demo. The other two demos cannot show an empty state because all available filtering options should have at least 1 results so I've left it out for those for now. Let me know if I should add it in anyways.

Added the indeterminate state in for the bulk select checkbox.

mcarrano reviewed (approved):
@kmcfaul Thanks for making the requested updates. I'll defer to @mmenestr on other questions.

kmcfaul commented:
@mmenestr 
This update should improve the table headers by giving them a bit more space. It appears that there's a small gap between when truncation is applied and when a tooltip is added (because making the columns thinner, or increasing the column name length both fix the issue). I can open up a follow up issue to investigate this later.

I don't really see a good option to fix the column widths though. It's trying to fit to content by default and when the table is filtered the collective cells' lengths are different. We have a `width` property but it only supports % widths and will still try to fit to content. @nicolethoen Any thoughts how this can be implemented?

kmcfaul commented:
Opened #7965 to follow up for the rest of the demos and other improvements

tlabaj commented:
Looks great Katie!  I have a question that is not a blocker.

When there are no matches, the empty state is show.  When this happens, the columns shift to take up the space of the checkbox column.  I am noticing the last column gets truncated 

<img width="886" alt="Screen Shot 2022-09-12 at 2 56 44 PM" src="https://user-images.githubusercontent.com/39532947/189734564-3269e07a-1801-464b-a03f-14b0267ae82f.png">

tlabaj reviewed (approved):
LGTM

nicolethoen commented:
> I don't really see a good option to fix the column widths though. It's trying to fit to content by default and when the table is filtered the collective cells' lengths are different. We have a `width` property but it only supports % widths and will still try to fit to content. @nicolethoen Any thoughts how this can be implemented?

I asked @mattnolting - 
Since the length of the cell text varies enough to cause column widths to change depending on the applied filter, the best thing is probably to pick the ideal column width which could be using the `width={x}` like you have now, or maybe using `modifier="fitContent"` (which will shrink the column to the size of the th so they will stay consistent).
In either case, you likely want to use the `modifier='truncate'` on the `Td` cells so that the longest values dont push columns around.

mmenestr reviewed (approved):
Looks great!! Thank you so much @kmcfaul

kmcfaul commented:
It should be fine because they render separately but I'll update them in the follow up as well
